src/time/Waveform.cpp

```diff
@@ -31,8 +32,17 @@ void Waveform::store(const Eigen::VectorXd &data)
   this->_timeWindows.col(columnID) = data;
 }
 
+Eigen::VectorXd Waveform::sample(double dt, int timeWindows, int order)
+{
+  PRECICE_ASSERT(dt >= 0, "Sampling outside of valid range!");
+  PRECICE_ASSERT(dt <= 1, "Sampling outside of valid range!");
+  // @ todo: Add more logic here? Set order in constructor; keep track of time windows inside class. See https://github.com/precice/precice/pull/1004/files?file-filters%5B%5D=.cmake&file-filters%5B%5D=.hpp#r642223767.
```

Comment:
yes, `order` should be a member variable in my opinion

Reply:
I will try to clean this up when implementing the configuration.

Reply:
Also done in https://github.com/precice/precice/pull/1083. I still have to implement this change here.

Reply:
Done in a0ce216.